packages/cli/src/metadataGeneration/metadataGenerator.ts

```diff
@@ -111,6 +112,71 @@ export class MetadataGenerator {
     }
   };
 
+  private checkForPathParamSignatureDuplicates = (controllers: Tsoa.Controller[]) => {
+    const controllerDup: { [key: string]: any } = {};
+    let message = '';
+
+    controllers.forEach(controller => {
+      const chunks: { [key: string]: Tsoa.Method[] } = {};
+      controller.methods.forEach(method => {
+        if (chunks[method.method] === undefined) {
+          chunks[method.method] = [];
+        }
+        chunks[method.method].push(method);
+      });
+
+      const chunkDupPath: { [key: string]: Tsoa.Method[] } = {};
+      const regex = new RegExp('^{.*?}/?');
+      Object.keys(chunks).forEach((key: string) => {
+        let containFirstPath = false;
+        const chunk = chunks[key];
+        const duplicates: Tsoa.Method[] = [];
+        chunk.forEach(value => {
+          const currentMatch = regex.test(value.path);
+          if (currentMatch) {
+            duplicates.push(value);
+          }
+          containFirstPath = containFirstPath || currentMatch;
+        });
+
+        if (duplicates.length > 1) {
+          chunkDupPath[key] = duplicates;
+        }
+      });
+
+      if (Object.keys(chunkDupPath).length > 0) {
+        controllerDup[controller.name] = chunkDupPath;
+      }
+    });
+
+    if (Object.keys(controllerDup).length > 0) {
+      message = `Duplicate path parameter definition signature found in controller `;
+      Object.keys(controllerDup).forEach((conKey, i, iArr) => {
+        message += `${conKey} at `;
+        const methodDup = controllerDup[conKey];
+        Object.keys(methodDup).forEach((methodKey, j, jArr) => {
+          message += `[method ${methodKey.toUpperCase()} `;
+          const dup = methodDup[methodKey];
+          dup.forEach((each: Tsoa.Method, k, kArr) => {
+            message += `${each.name}`;
+            if (k !== kArr.length - 1) {
+              message += ', ';
+            }
+          });
+          message += j !== jArr.length - 1 ? '], ' : ']';
+        });
+        if (i !== iArr.length - 1) {
+          message += ', ';
+        }
+      });
```

Comment:
IMO you could use `map` and `join` for more readability:
```ts
message += Object.keys(controllerDup).map((conKey) => {
	const keys = Object.keys(controllerDup[conKey]);
	return `${conKey} at [${keys.map((methodKey) => {
		const dup = methodDup[methodKey];
		return `method ${methodKey.toUpperCase()} ${dup.map((each: Tsoa.Method) => {
			return each.name;
		}).join(', ')}`;
	}).join('], [')}]`;
}).join(', ');
```

Note: maybe there is a typo in my code but you've got the point

Reply:
Thank you! It does help a lot with concatenation!